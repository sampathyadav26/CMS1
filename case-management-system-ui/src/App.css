.custom-textfield {
  height: 5px !important;
  /* width: px !important; */
  /* border-radius: 7px;
  border-width: 1px; */
  padding: 2px !important;
}

.rs-input-group.rs-input-group-inside {
  height: 30px;
}

.box {
  margin-left: 20%;
  height: 80%;
}
.box1 {
  margin-left: 20%;
  margin-top: 10%;
}

.ag-theme-alpine .ag-root-wrapper {
  border: none !important;
}

.ag-theme-alpine .ag-header-container {
  border-bottom: none !important;
}
.ag-theme-alpine .ag-row {
  border-bottom: none !important;
}
.ag-theme-alpine .ag-header-cell,
.ag-theme-alpine .ag-header-group-cell {
  background-color: #e9eef4 !important;
  color: #1b4f91 !important; /* Change text color to white for better readability */
}
.ag-theme-alpine .ag-root-wrapper {
  border-radius: 10px; /* Adjust as needed */
  overflow: hidden; /* This ensures inner parts do not overflow the rounded corners */
}
.ag-theme-alpine .ag-header-container {
  border-bottom: none !important;
}
.ag-theme-alpine .ag-row {
  border-bottom: none !important;
}

.ag-theme-alpine .ag-body-viewport::-webkit-scrollbar {
  width: 4px;
}

.ag-theme-alpine .ag-body-viewport::-webkit-scrollbar-thumb {
  background-color: black;
  border-radius: 61px;
}

.ag-theme-alpine .ag-body-viewport::-webkit-scrollbar-track {
  background-color: #888;
}
.rs-picker-popup {
  z-index: 9999 !important;
}
