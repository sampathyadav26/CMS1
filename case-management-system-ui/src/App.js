import React, { useEffect, Suspense } from "react";
import CustomRoutes from "./router";
import { BrowserRouter as Router } from "react-router-dom";
import { useDispatch } from "react-redux";
import { onActiveMobileView } from "./store/actions";
import { isMobile } from "react-device-detect";

const App = () => {
  const dispatch = useDispatch();
  const handleWindowSizeChange = () => {
    const isMobileView = window.innerWidth <= 767;
    console.info("handleWindowSizeChange called", isMobileView);
    dispatch(onActiveMobileView(isMobileView || isMobile));
  };
  const handleContextmenu = (ev) => {
    console.info("ev ##", ev);
  };
  useEffect(() => {
    window.addEventListener("resize", handleWindowSizeChange);
    window.addEventListener("contextmenu", handleContextmenu);
    return () => {
      window.removeEventListener("resize", handleWindowSizeChange);
      window.removeEventListener("contextmenu", handleContextmenu);
    };
  }, []);
  console.info("process.env ##", process.env);
  return (
    <>
      <Router>
        <Suspense>
          <CustomRoutes />
        </Suspense>
      </Router>
    </>
  );
};

export default App;
