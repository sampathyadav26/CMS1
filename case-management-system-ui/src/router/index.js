import React, { lazy, Suspense } from "react";
import { Navigate, Routes, Route, useNavigate } from "react-router-dom";
import { useIdleTimer } from "react-idle-timer";
import PageNotFound from "../components/common/PageNotFound";



import Login from "../components/common/Login";
import Register from "../components/common/Registration";
import LandingPage from "../components/landingPage";

import CaseIdView from "../components/dashboard/caseIdView";
import AddCase from "../components/dashboard/addCase";

import CustomerDetailsView from "../components/customer/customerDetailsView";
import CreateCustomer from "../components/customer/createCustomer";

const Dashboard = lazy(() => import("../components/dashboard"));
const CustComponent = lazy(() => import("../components/customer"));

const ResetPassword = lazy(() => import("../components/common/ResetPassword"));

const ForgotPasswordPage = lazy(() =>
  import("../components/common/ForgotPasswordPage")
);

const PrivateRoute = lazy(() => import("../PrivateRoute"));


const CustomRoutes = () => {
  const nav = useNavigate();
  useIdleTimer({
    timeout: 1000 * 60 * 15, // time in milliseconds (15 Mins)
    onIdle: () => {
      nav("/cmsapp");
    },
    debounce: 500,
  });
  return (
    <Suspense>
      <Routes>
        <Route exact path="/" element={<LandingPage />}>
          <Route
            path="/cmsapp/dashboard"
            element={<PrivateRoute component={Dashboard} />}
          />
          <Route
            path="/cmsapp/customer"
            element={<PrivateRoute component={CustComponent} />}
          /> 
        
        


          <Route
            exact
            path="/cmsapp/"
            element={<Navigate to={"/cmsapp/dashboard"} />}
          />
          <Route
            path="/cmsapp/caseIdView"
            element={<PrivateRoute component={CaseIdView} />}
          />
          <Route
            path="/cmsapp/createCase"
            element={<PrivateRoute component={AddCase} />}
          />
          <Route
            path="/cmsapp/createCustomer"
            element={<PrivateRoute component={CreateCustomer} />}
          />
          <Route
            path="/cmsapp/customerIdView"
            element={<PrivateRoute component={CustomerDetailsView} />}
          />
          <Route path="/cmsapp/customer" />
          <Route path="*" element={<Navigate to="/cmsapp/404" />} />
          <Route path="/cmsapp/404" element={<PageNotFound />} />
        </Route>
        <Route path="/cmsapp/register" element={<Register />} />
        <Route
          path="/cmsapp/forgotPasswordPage"
          element={<ForgotPasswordPage />}
        />
        <Route path="/cmsapp/resetPassword" element={<ResetPassword />} />
        <Route path="/cmsapp/login" element={<Login />} />
      </Routes>
    </Suspense>
  );
};
export default CustomRoutes;
