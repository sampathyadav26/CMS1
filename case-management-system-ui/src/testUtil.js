/* istanbul ignore file*/
import React from "react";
import { Provider } from "react-redux";
import {
  render,
  within,
  waitForElementToBeRemoved,
} from "@testing-library/react";
import UserEvent from "@testing-library/user-event";
import { BrowserRouter as Router } from "react-router-dom";
import { setupStore } from "./store";
const confStore = setupStore();

export const renderWithOutRouter = (component, { store = confStore } = {}) => {
  return {
    ...render(<Provider store={store}>{component}</Provider>),
    store,
  };
};

const renderWithRedux = (component, { store = confStore } = {}) => {
  return {
    ...render(
      <Provider store={store}>
        <Router>{component}</Router>
      </Provider>
    ),
    store,
  };
};
export default renderWithRedux;

export const renderWithProviders = (
  ui,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = setupStore(preloadedState),
    ...renderOptions
  } = {}
) => {
  const Wrapper = ({ children }) => {
    return (
      <Provider store={store}>
        <Router>{children}</Router>
      </Provider>
    );
  };
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
};

//selectMaterialUiSelectOption
export const selectMaterialUiSelectOption = async (element, optionText) =>
  new Promise((resolve) => {
    // The the button that opens the dropdown, which is a sibling of the input
    const selectButton = element.parentNode.querySelector("[role=button]");

    // Open the select dropdown
    UserEvent.click(selectButton);

    // Get the dropdown element. We don't use getByRole() because it includes <select>s too.
    const listbox = document.body.querySelector("ul[role=listbox]");

    // Click the list item
    const listItem = within(listbox).getByText(optionText);
    UserEvent.click(listItem);

    // Wait for the listbox to be removed, so it isn't visible in subsequent calls
    waitForElementToBeRemoved(() =>
      document.body.querySelector("ul[role=listbox]")
    ).then(resolve);
  });
