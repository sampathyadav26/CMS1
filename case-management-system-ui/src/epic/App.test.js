import App from "../App";
import { renderWithOutRouter } from "../testUtil";
import { cleanup } from "@testing-library/react";
import "../router";

afterEach(cleanup);

test("Render App container", () => {
  const container = renderWithOutRouter(<App />, {});
  expect(container).toMatchSnapshot();
});
