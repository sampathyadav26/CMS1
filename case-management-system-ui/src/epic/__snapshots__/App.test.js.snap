// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App container 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 sc-aXZVg kFNeUm css-1ps6pg7-MuiPaper-root"
          style="height: 100vh;"
        >
          <div
            class="sc-gEvEer eyhQuh"
          >
            <div
              class="sc-eqUAAy TknXt"
              style="display: flex; align-items: center;"
            >
              <div
                class="sc-fqkvVR eijmdp"
              >
                <img
                  alt="Logo"
                  src="CMS_logo.png"
                  style="height: 60px; width: 60px; margin-right: 5px;"
                />
                <img
                  alt="LogoShadow"
                  id="shawdow"
                  src="CMS_logo_shadow.png"
                  style="height: 60px; width: 60px;"
                />
              </div>
              <div
                class="sc-dcJsrY iyNfke"
              >
                <h5
                  class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter css-1ewk6ns-MuiTypography-root"
                >
                  Case Management System
                </h5>
                <p
                  class="MuiTypography-root MuiTypography-body2 MuiTypography-alignLeft css-1jo1j2q-MuiTypography-root"
                >
                  Navigating Challenges, Building Solutions
                </p>
              </div>
            </div>
          </div>
          <div
            class="sc-iGgWBj dyfnJm MuiBox-root css-0"
          >
            <div
              class="box1 MuiBox-root css-0"
            >
              <h4
                class="MuiTypography-root MuiTypography-h4 css-zij4st-MuiTypography-root"
                style="font-family: Inter; font-size: 20px; font-weight: 500; line-height: 24.2px; letter-spacing: 0em; text-align: left;"
              >
                Welcome !
              </h4>
              <p
                class="MuiTypography-root MuiTypography-body1 css-1vw91mc-MuiTypography-root"
              >
                User ID*
              </p>
              <div
                class="MuiFormControl-root MuiTextField-root custom-textfield css-1js2rtn-MuiFormControl-root-MuiTextField-root"
              >
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                    id="email"
                    name="email"
                    placeholder="Email"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-ihdtdm"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <p
                class="MuiTypography-root MuiTypography-body1 css-eof58q-MuiTypography-root"
              >
                Password*
              </p>
              <div
                class="MuiFormControl-root MuiTextField-root custom-textfield css-8zg2bs-MuiFormControl-root-MuiTextField-root"
              >
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                    id="password"
                    name="password"
                    placeholder="*******"
                    type="password"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-ihdtdm"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <div
                class="MuiGrid-root css-vj1n65-MuiGrid-root"
                style="margin-top: 4%;"
              >
                <a
                  href="/cmsapp/forgotPasswordPage"
                  style="font-family: Inter; font-size: 14px; font-weight: 400; line-height: 17px; letter-spacing: 0em; text-align: left;"
                  sx="[object Object]"
                >
                  Forgot password?
                </a>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-13i4rnv-MuiGrid-root"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-wgejo3-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="submit"
                >
                  Login
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-13i4rnv-MuiGrid-root"
              >
                <a
                  href="/cmsapp/register"
                  sx="[object Object]"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1ovj7jt-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="button"
                  >
                    Sign up
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </a>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 sc-aXZVg kFNeUm css-1ps6pg7-MuiPaper-root"
        style="height: 100vh;"
      >
        <div
          class="sc-gEvEer eyhQuh"
        >
          <div
            class="sc-eqUAAy TknXt"
            style="display: flex; align-items: center;"
          >
            <div
              class="sc-fqkvVR eijmdp"
            >
              <img
                alt="Logo"
                src="CMS_logo.png"
                style="height: 60px; width: 60px; margin-right: 5px;"
              />
              <img
                alt="LogoShadow"
                id="shawdow"
                src="CMS_logo_shadow.png"
                style="height: 60px; width: 60px;"
              />
            </div>
            <div
              class="sc-dcJsrY iyNfke"
            >
              <h5
                class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter css-1ewk6ns-MuiTypography-root"
              >
                Case Management System
              </h5>
              <p
                class="MuiTypography-root MuiTypography-body2 MuiTypography-alignLeft css-1jo1j2q-MuiTypography-root"
              >
                Navigating Challenges, Building Solutions
              </p>
            </div>
          </div>
        </div>
        <div
          class="sc-iGgWBj dyfnJm MuiBox-root css-0"
        >
          <div
            class="box1 MuiBox-root css-0"
          >
            <h4
              class="MuiTypography-root MuiTypography-h4 css-zij4st-MuiTypography-root"
              style="font-family: Inter; font-size: 20px; font-weight: 500; line-height: 24.2px; letter-spacing: 0em; text-align: left;"
            >
              Welcome !
            </h4>
            <p
              class="MuiTypography-root MuiTypography-body1 css-1vw91mc-MuiTypography-root"
            >
              User ID*
            </p>
            <div
              class="MuiFormControl-root MuiTextField-root custom-textfield css-1js2rtn-MuiFormControl-root-MuiTextField-root"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="email"
                  name="email"
                  placeholder="Email"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-ihdtdm"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <p
              class="MuiTypography-root MuiTypography-body1 css-eof58q-MuiTypography-root"
            >
              Password*
            </p>
            <div
              class="MuiFormControl-root MuiTextField-root custom-textfield css-8zg2bs-MuiFormControl-root-MuiTextField-root"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="password"
                  name="password"
                  placeholder="*******"
                  type="password"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-ihdtdm"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiGrid-root css-vj1n65-MuiGrid-root"
              style="margin-top: 4%;"
            >
              <a
                href="/cmsapp/forgotPasswordPage"
                style="font-family: Inter; font-size: 14px; font-weight: 400; line-height: 17px; letter-spacing: 0em; text-align: left;"
                sx="[object Object]"
              >
                Forgot password?
              </a>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-13i4rnv-MuiGrid-root"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-wgejo3-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="submit"
              >
                Login
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-13i4rnv-MuiGrid-root"
            >
              <a
                href="/cmsapp/register"
                sx="[object Object]"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1ovj7jt-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                >
                  Sign up
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </a>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
